dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/saml/SamlGroupService.java

```diff
@@ -0,0 +1,92 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Stateless
+public class SamlGroupService {
+    private static final Logger logger = LoggerFactory.getLogger(SamlGroupService.class);
+
+    private RoleAssigneeServiceBean roleAssigneeSvc;
+
+    private ActionLogServiceBean actionLogSvc;
+
+    private SamlGroupRepository repository;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public SamlGroupService() { }
+
+    @Inject
+    public SamlGroupService(RoleAssigneeServiceBean roleAssigneeSvc, ActionLogServiceBean actionLogSvc, SamlGroupRepository repository) {
+        this.roleAssigneeSvc = roleAssigneeSvc;
+        this.actionLogSvc = actionLogSvc;
+        this.repository = repository;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public List<SamlGroup> findAll() {
+        return repository.findAll();
+    }
+
+    public SamlGroup findById(Long id) {
+        return repository.findById(id).orElse(null);
+    }
+
+    public SamlGroup save(String name, String entityId) {
+        ActionLogRecord alr = new ActionLogRecord(ActionLogRecord.ActionType.GlobalGroups, "shibCreate");
+        alr.setInfo(name + ": " + entityId);
+
+        SamlGroup saved = repository.saveAndFlush(new SamlGroup(name, entityId));
+        actionLogSvc.log(alr);
+        return saved;
+    }
+
+    public Set<SamlGroup> findFor(AuthenticatedUser authenticatedUser) {
+        Set<SamlGroup> groupsForUser = new HashSet<>();
+        String samlIdPEntityId = authenticatedUser.getSamlIdPEntityId();
+        logger.debug("IdP for user " + authenticatedUser.getIdentifier() + " is " + samlIdPEntityId);
+        if (samlIdPEntityId != null) {
+            groupsForUser.addAll(repository.findByEntityId(samlIdPEntityId));
+        }
+        return groupsForUser;
+    }
+
+    public boolean delete(SamlGroup doomed) throws Exception {
+        ActionLogRecord alr = new ActionLogRecord(ActionLogRecord.ActionType.GlobalGroups, "samlDelete");
+        alr.setInfo(doomed.getName() + ":" + doomed.getIdentifier());
+
+        List<RoleAssignment> assignments = roleAssigneeSvc.getAssignmentsFor(doomed.getIdentifier());
+        if (assignments.isEmpty()) {
+            repository.mergeAndDelete(doomed);
+            actionLogSvc.log(alr);
+            return true;
+        } else {
+            List<String> assignmentIds = new ArrayList<>();
+            for (RoleAssignment assignment : assignments) {
+                assignmentIds.add(assignment.getId().toString());
+            }
+            String message = String.format("Could not delete Saml group with id [%d] due to existing role assignments: %s",
+                    doomed.getId(), assignmentIds);
+            logger.info(message);
+            actionLogSvc.log(alr.setActionResult(ActionLogRecord.Result.BadRequest)
+                                     .setInfo(alr.getInfo() + "// " + message));
+            throw new Exception(message);
```

Comment:
Can we pick more suitable exception? When we catch this Exception in api we are basically are catching everything